app/src/test/java/org/mozilla/focus/search/SearchEngineWriterTest.kt

```diff
@@ -0,0 +1,38 @@
+package org.mozilla.focus.search
+
+import android.graphics.Bitmap
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+import org.junit.Assert.assertEquals
+
+@RunWith(RobolectricTestRunner::class)
+class SearchEngineWriterTest {
+
+    @Test
+    fun buildSearchEngineXML() {
+        val endSearchTermXml = "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>" +
+            "<OpenSearchDescription xmlns=\"http://a9.com/-/spec/opensearch/1.1/\">" +
+            "<ShortName>testEngine</ShortName>" +
+            "<Image height=\"16\" width=\"16\">data:image/png;base64,Qml0bWFwICgxMCB4IDEwKSBjb21wcmVzc2VkIGFzIFBORyB3aXRoIHF1YWxpdHkgMTAw\n</Image>" +
+            "<Description>testEngine</Description>" +
+            "<Url template=\"testEngine/?q={searchTerms}\" type=\"text/html\"/>" +
+            "</OpenSearchDescription>"
+
+        assertEquals(endSearchTermXml,
+            SearchEngineWriter.buildSearchEngineXML("testEngine", "testEngine/?q=%s", createBitmap()))
+
+        val middleSearchTermXml = "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>" +
+            "<OpenSearchDescription xmlns=\"http://a9.com/-/spec/opensearch/1.1/\">" +
+            "<ShortName>testEngine</ShortName>" +
+            "<Image height=\"16\" width=\"16\">data:image/png;base64,Qml0bWFwICgxMCB4IDEwKSBjb21wcmVzc2VkIGFzIFBORyB3aXRoIHF1YWxpdHkgMTAw\n</Image>" +
+            "<Description>testEngine</Description>" +
+            "<Url template=\"testEngine/?q={searchTerms}/tests\" type=\"text/html\"/>" +
+            "</OpenSearchDescription>"
+
+        assertEquals(middleSearchTermXml,
+            SearchEngineWriter.buildSearchEngineXML("testEngine", "testEngine/?q=%s/tests", createBitmap()))
+    }
+
+    private fun createBitmap() = Bitmap.createBitmap(10, 10, Bitmap.Config.ARGB_8888)
```

Comment:
\[detekt\] INFO: \[style/MagicNumber\] Report magic numbers\. Magic number is a numeric literal that is not defined as a constant and hence it's unclear what the purpose of this number is\. It's better to declare such numbers as constants and give them a proper name\. By default, \-1, 0, 1, and 2 are not considered to be magic numbers\.